### DISCUSSION TITLE
Issue with setting AI-Deck to connect with WiFi AP

### DISCUSSION THREAD

**Discussion by Kaijun101** [bitcraze/discussions#509] [Q&A] [answered]
Hi, I am currently having issues with setting up the AI-Deck to connect to WiFi, i.e., after flashing the cf firmware the CF does not connect to the wifi or rather stays as an access point. The CF is currently using the 2022.12 firmware. I have followed the steps in : https://github.com/bitcraze/aideck-gap8-examples/blob/master/docs/test-functions/wifi-streamer.md
```
$ git clone https://github.com/bitcraze/crazyflie-firmware.git
$ cd crazyflie-firmware
$ make menuconfig
$ make
$ cfloader flash cf2.bin stm32-fw -w [CRAZYFLIE_URI]
```
Below are the output from the console:
```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:b8d2e0b9e983 (2022.12) CLEAN
SYS: I am 0x3530343431315103001C002B and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 2 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow for deck 0
PMW: Motion chip id: 0x49:0xB6
DECK_CORE: Calling INIT on driver bcAI for deck 1
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized
AIDECK: Not setting up WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.12
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
DECK_CORE: Deck 1 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 11712 bytes
CPX: ESP32: I (800) SYS: Initialized
CPX: ESP32: I (800) DISCOVERY: Hostname is aideck-E0CA5C
CPX: ESP32: I (800) DISCOVERY: MAC (STA) is 9c:9c:1f:e0:ca:5c
STAB: Ready to fly.
```

**Comment by hmllr** [comment, marked as answer]
I am sorry, but I am not sure what you are exactly trying to do - do you want the AI-deck to act as an access point or to connect to an existing wifi? The readme is for acting as an access point, if you want it to connect to an existing wifi you will have to configure it differently. Can you also send your config file? (should be saved in the build folder as .config per default)

Best,
Hanna

**Reply by Kaijun101** [reply]
Hi @hmllr ,

I am trying to get AI-Deck to connect to a existing wifi network. My config file is as such:
```
#
# Automatically generated file; DO NOT EDIT.
# Crazyflie Platform Configuration
#

#
# Build and debug options
#
CONFIG_CROSS_COMPILE="arm-none-eabi-"
# CONFIG_DEBUG is not set
# CONFIG_DEBUG_ENABLE_LED_MORSE is not set
# CONFIG_DEBUG_PRINT_ON_UART1 is not set
# CONFIG_DEBUG_DECK_IGNORE_OW is not set

#
# Platform configuration
#
CONFIG_PLATFORM_CF2=y
# CONFIG_PLATFORM_BOLT is not set
# CONFIG_PLATFORM_TAG is not set
# CONFIG_PLATFORM_FLAPPER is not set

#
# IMU configuration
#
# CONFIG_IMU_MADGWICK_QUATERNION is not set
CONFIG_IMU_MAHONY_QUATERNION=y

#
# Sensor configuration
#
CONFIG_SENSORS_MPU9250_LPS25H=y
CONFIG_SENSORS_BMI088_BMP388=y
# CONFIG_SENSORS_BOSCH is not set
# CONFIG_SENSORS_IGNORE_BAROMETER_FAIL is not set
# CONFIG_SENSORS_BMI088_SPI is not set
CONFIG_SENSORS_BMI088_I2C=y

#
# Power management configuration
#
# CONFIG_PM_AUTO_SHUTDOWN is not set

#
# Communication
#
# CONFIG_SYSLINK_RX_DMA is not set
CONFIG_ENABLE_CPX=y
CONFIG_ENABLE_CPX_ON_UART2=y
CONFIG_CPX_UART2_BAUDRATE=576000

#
# App layer configuration
#
# CONFIG_APP_API_TEST is not set
# CONFIG_APP_ENABLE is not set

#
# Expansion deck configuration
#
CONFIG_DECK_FORCE="none"
CONFIG_DECK_ACTIVE_MARKER=y
CONFIG_DECK_AI=y
# CONFIG_DECK_AI_WIFI_NO_SETUP is not set
# CONFIG_DECK_AI_WIFI_SETUP_AP is not set
CONFIG_DECK_AI_WIFI_SETUP_STA=y

#
# Credentials for access-point
#
CONFIG_DECK_AI_SSID="nvidia"
CONFIG_DECK_AI_PASSWORD="password123"
# CONFIG_DECK_BIGQUAD is not set
CONFIG_DECK_BUZZ=y
# CONFIG_DECK_CPPM is not set
# CONFIG_DECK_FLAPPER is not set
CONFIG_DECK_FLOW=y
# CONFIG_DECK_GTGPS is not set
CONFIG_DECK_LEDRING=y
CONFIG_DECK_LEDRING_DEFAULT_EFFECT=6
CONFIG_DECK_LEDRING_NBR_LEDS=12
CONFIG_DECK_LEDRING_DIMMER=0
CONFIG_DECK_LIGHTHOUSE=y
# CONFIG_DECK_LIGHTHOUSE_AS_GROUNDTRUTH is not set
CONFIG_DECK_LIGHTHOUSE_MAX_N_BS=4
CONFIG_DECK_LOCO=y
# CONFIG_DECK_LOCODECK_USE_ALT_PINS is not set
CONFIG_DECK_LOCO_NR_OF_ANCHORS=8
CONFIG_DECK_LOCO_ALGORITHM_AUTO=y
# CONFIG_DECK_LOCO_ALGORITHM_TWR is not set
# CONFIG_DECK_LOCO_ALGORITHM_TDOA2 is not set
# CONFIG_DECK_LOCO_ALGORITHM_TDOA3 is not set
# CONFIG_DECK_LOCO_FULL_TX_POWER is not set
CONFIG_DECK_MULTIRANGER=y
CONFIG_DECK_OA=y
CONFIG_DECK_USD=y
# CONFIG_DECK_USD_USE_ALT_PINS_AND_SPI is not set
CONFIG_DECK_ZRANGER=y
CONFIG_DECK_ZRANGER2=y
# CONFIG_DECK_CPX_HOST_ON_UART2 is not set

#
# Controllers and Estimators
#
CONFIG_CONTROLLER_ANY=y
# CONFIG_CONTROLLER_PID is not set
# CONFIG_CONTROLLER_INDI is not set
# CONFIG_CONTROLLER_MELLINGER is not set
# CONFIG_CONTROLLER_BRESCIANINI is not set
CONFIG_ESTIMATOR_KALMAN_ENABLE=y
CONFIG_ESTIMATOR_ANY=y
# CONFIG_ESTIMATOR_KALMAN is not set
# CONFIG_ESTIMATOR_COMPLEMENTARY is not set
# CONFIG_ESTIMATOR_OOT is not set

#
# Motor configuration
#
# CONFIG_MOTORS_ESC_PROTOCOL_STANDARD_PWM is not set
CONFIG_MOTORS_ESC_PROTOCOL_ONESHOT125=y
# CONFIG_MOTORS_ESC_PROTOCOL_ONESHOT42 is not set
# CONFIG_MOTORS_ESC_PROTOCOL_DSHOT is not set
# CONFIG_MOTORS_START_DISARMED is not set
CONFIG_MOTORS_DEFAULT_PROP_TEST_PWM_RATIO=0
CONFIG_POWER_DISTRIBUTION_QUADROTOR=y
CONFIG_ENABLE_THRUST_BAT_COMPENSATED=y

#
# Parameter subsystem
#
# CONFIG_PARAM_SILENT_UPDATES is not set
```

**Reply by hmllr** [reply]
the config file actually looks correct. I would try to update all firmwares again to the newest release - meaning the STM32, the nRF, the NINA and best also the gap8. You can follow the steps in https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade for the nRF and NINA, but be aware that it also overwrites the STM32 firmware so you'll have to flash it again afterwards.
I hope this helps (for me it works, I just tested it),
Hanna

**Reply by Kaijun101** [reply]
Hi @hmllr.

After following the steps in the link you sent, I noticed that the STM32 and nRF is being update with 2022.12 but the NINA has no indications of being updated. It still does not connect to the wifi.

I then proceed with the following commands:
```
$ git clone https://github.com/bitcraze/crazyflie-firmware.git
$ cd crazyflie-firmware
$ make menuconfig
$ make
$ cfloader flash cf2.bin stm32-fw -w [CRAZYFLIE_URI]
```
Below is the output I get from the CF client:
```
SYS: Crazyflie 2.1 is up and running!
SYS: Build 19:af96c070d6bc (2022.12 +19) CLEAN
SYS: I am 0x203235324133501900510034 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 3 deck(s) found
DECK_CORE: Calling INIT on driver bcMultiranger for deck 0
DECK_CORE: Calling INIT on driver bcFlow2 for deck 1
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
DECK_CORE: Calling INIT on driver bcAI for deck 2
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized
AIDECK: AI-deck will connect to WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.12
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
MR: Init front sensor [OK]
MR: Init back sensor [OK]
MR: Init up sensor [OK]
MR: Init left sensor [OK]
MR: Init right sensor [OK]
DECK_CORE: Deck 0 test [OK].
DECK_CORE: Deck 1 test [OK].
DECK_CORE: Deck 2 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 10400 bytes
STAB: Ready to fly.
```

**Reply by hmllr** [reply]
Now you get the "AIDECK: AI-deck will connect to WiFi" contrary to the "AIDECK: Not setting up WiFi", so that's already much better, it seems as if now the config on the STM32 are correct. However, I believe the NINA update is the problem - did you try the update with only the AI-deck connected? No other deck, that's important, otherwise it will not work.

**Reply by Kaijun101** [reply]
So I upgrade the CF through the CF client with only the AI-Deck. Then I make the firmware and uploaded it again. However, it seems like it still does not connect to the wifi.

```
SYS: Crazyflie 2.1 is up and running!
SYS: Build 19:af96c070d6bc (2022.12 +19) CLEAN
SYS: I am 0x203235324133501900510034 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcAI for deck 0
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized
AIDECK: AI-deck will connect to WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.12
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 14600 bytes
CPX: ESP32: I (801) SYS: Initialized
CPX: ESP32: I (801) DISCOVERY: Hostname is aideck-55F2F8
CPX: ESP32: I (801) DISCOVERY: MAC (STA) is ec:94:cb:55:f2:f8
CPX: ESP32: I (2731) WIFI: wifi_init_sta finished.
STAB: Ready to fly.
CPX: ESP32: I (4061) WIFI: wifi_init_softap finished
```

**Reply by hmllr** [reply]
looks already way better! you now get the CPX packets :) Are you sure your computer is connected to the same wifi you have the credentials in the config file for? the output all looks good to me, it is just missing the end:
![Screenshot from 2023-01-10 09-35-23](https://user-images.githubusercontent.com/14876009/211501239-bb8e0424-18bc-4a23-90f4-2e286e069f45.png)

**Reply by Kaijun101** [reply]
Yes, my computer connected to the same network. I noticed that wifi streaming example is still in my list of wifi but now it has a password which corresponds to my local wifi passsword.

**Reply by hmllr** [reply]
hmm, did you also update the gap8? as described in the readme of the wifi streamer example? (if you have an older AI deck you might have to flash the bootloader first yourself)

**Reply by Kaijun101** [reply]
I am sorry do you means the steps in this https://github.com/bitcraze/aideck-gap8-examples/blob/master/docs/test-functions/wifi-streamer.md? i do not see any instructions on updating the GAP8, so lost as in what you mean by updating the GAP8. But, I have flash the examples as shown.

**Reply by hmllr** [reply]
yes, I mean the updating and flashing part (https://github.com/bitcraze/aideck-gap8-examples/blob/master/docs/test-functions/wifi-streamer.md#building-and-flashing-the-example). So this finished successfully?

**Reply by Kaijun101** [reply]
Yes, I think so. I can connect to the wifi streaming example network of nina and run the opencv-viewer. Not sure why this network persist on the AI-deck.

**Reply by hmllr** [reply]
so do you reach 100% when executing the commands above? I am not 100% sure, but I think in an older firmware of the gap8 it was the gap8 that sent the wifi setup command to the NINA (and not the STM32 as now), so my theory is that if there is an old firmware on the gap8 it might send the cmd to set up an access point to the NINA. Which then is most likely faster than the command from the STM32 (because that one has a lot more tasks to start up first), so the NINA will already open an access point.

**Reply by matthewoots** [reply]
@hmllr Hi there, so is there any way to flash the gap8 through the `cfloader` or the only way is the flash the gap8 using the jtag? and if so which firmware should we use to flash? And do we need to set it in bootloader mode?

**Reply by Kaijun101** [reply]
Hi @hmllr, actually when setting up the docker, I could not receive a link for the auto tiler, GreenWaves seems to no longer send link. So I did not finish that step. I will retry all the steps again. But the url step is not working at the moment.

Also, just asking, if next time I flash my code on gap8, will the wifi stop working again?

**Reply by hmllr** [reply]
@matthewoots It depends how new your board is - for 1.0 and 1.1 rev.A-C you need to flash the bootloader once and then you can use the over the air flashing. If you have a rev. D the bootloader should be already flashed. you find more information here: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/
you don't need to set it in bootloader mode manually, the cfloader will do this for you.

**Reply by hmllr** [reply]
> Hi @hmllr, actually when setting up the docker, I could not receive a link for the auto tiler, GreenWaves seems to no longer send link. So I did not finish that step. I will retry all the steps again. But the url step is not working at the moment.
> 
> Also, just asking, if next time I flash my code on gap8, will the wifi stop working again?

you should not need the autotiler for flashing the bootloader and the WiFi streamer example, only for the classifier. But we will investigate the issue with no link sent.

What do you mean by flash your code on the gap8? it depends what your code does, but in general the wifi should still work.

**Reply by Kaijun101** [reply]
Hi @hmllr, after redoing all the steps again, the wifi finally gets connected again. Thanks a lot for your help! 

I mean like deploy my code on AI-Deck, because I see that a new bin is needed to be flash on the AI-Deck and wonder if it will wipe the wifi settings.

And yes please, do help to investigate the no email sent for the autotiler sign up.

**Reply by knmcguire** [reply]
We have Greenwaves know about this issue but also please contact Greenwaves yourself about the problem of not getting the autotiler signup. It might have also gone to your spamfolder perhaps.

The wifi settings are settled are set in the Crazyflie Firmware with Kbuild, so reflashing the AIdeck should not affect this.

**Reply by Yaooooo** [reply]
@Kaijun101 seems there's an issue on our side, my colleague will take a look. 
Before that will be fixed, you can also write an email to us: support@greenwaves-technologies.com
With your name, county and organization info, then we can send you the lib. 

Sorry for this.

**Reply by Kaijun101** [reply]
@Yaooooo thanks for the heads up, will sent an email to you.

**Comment by crazyflie-aljanabi** [comment]
Hie,

I am writing to inquire about how to enable the Access Point mode on my AI-deck in order to connect it with GAP8 via Wifi. My goal is to establish video transmission from the GAP8 camera to the Crazyflie client.

I would appreciate it if you could provide me with instructions or guidance on how to enable the Access Point mode on the AI-deck. Additionally, if there are any specific configurations or settings required for connecting with GAP8, please let me know.

Thank you in advance for your assistance.

Best regards,
Ali

**Reply by hmllr** [reply]
This question is not related to the original question, please start a new question.
